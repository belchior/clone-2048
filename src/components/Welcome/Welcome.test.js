import React from 'react';
import rerender from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Welcome } from './index';

it('Welcome should render with an empty list without crashing', () => {
  const wall = [];
  const button = { text: '', action: () => {}};

  const tree = rerender.create(
    <Welcome wall={wall} button={button}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('Welcome should render with a list without crashing', () => {
  const wall = [2,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0];
  const button = { text: 'Action', action: () => {}};

  const tree = rerender.create(
    <Welcome wall={wall} button={button}/>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('WelcomeAction button when clicked should call buttonAction callback', () => {
  const wall = [];
  const buttonAction = jest.fn();
  const button = { text: 'Action', action: buttonAction};

  const component = shallow(<Welcome wall={wall} button={button}/>);
  component.find('.WelcomeAction').simulate('click');

  expect(buttonAction.mock.calls).toHaveLength(1);
});
