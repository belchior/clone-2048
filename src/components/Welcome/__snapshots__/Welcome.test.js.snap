// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should render with a list without crashing 1`] = `
<div
  className="Welcome"
>
  <div
    className="Wall "
  >
    <div
      className="Block block-2"
    >
      2
    </div>
    <div
      className="Block block-2"
    >
      2
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
    <div
      className="Block block-0"
    >
      0
    </div>
  </div>
  <button
    className="WelcomeAction"
    onClick={[Function]}
    type="button"
  >
    Action
  </button>
</div>
`;

exports[`Welcome should render with an empty list without crashing 1`] = `
<div
  className="Welcome"
>
  <div
    className="Wall "
  />
  <button
    className="WelcomeAction"
    onClick={[Function]}
    type="button"
  >
    
  </button>
</div>
`;
